tasks/init.rb

```diff
@@ -1,12 +1,15 @@
+#!/opt/puppetlabs/puppet/bin/ruby
 require 'facter'
 require 'json'
 
 params   = JSON.parse(STDIN.read)
 timeout  = params['timeout'].to_i || 3
 message  = params['message']
 
-def async_command(cmd)
-  wait_time = 3
+# Force a minimum timeout of 3 seconds to allow the task response to be delivered.
+timeout = 3 if timeout < 3
```

Comment:
Should this be surfaced to the user at all? Since it's being used in a task I'm not sure how useful that'd be, but it seems a bit "magic" if folks have the timeout set less than 3.

Reply:
I'll add it to the task metadata. It's also part of the result of the task.